Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -173,6 +194,365 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+    StructuredArrow.mk (Y := WalkingParallelPair.one) (id zero)
+  intro x
+  have h : Inhabited (StructuredArrow x forgetReflexion)
+  · constructor
+    cases x with
+    | one => exact e₀
+    | zero => exact e₁
+  cases x with
+    | zero => apply IsConnected.of_induct (j₀ := e₁)
+              intro p h₁ h₂ t
+              rcases t with ⟨l, y, f⟩
+              cases y <;> cases f
+              · set r : StructuredArrow zero forgetReflexion :=
+                  StructuredArrow.mk (Y := WalkingParallelPair.zero) reflexion
+                change r ∈ p
+                suffices f : r ⟶  e₁ by exact (h₂ f).mpr h₁
+                exact StructuredArrow.homMk (WalkingParallelPairHom.left)
+              · exact h₁
+    | one => apply IsConnected.of_induct (j₀ := e₀)
+             intro p h₁ h₂ t
+             rcases t with ⟨l, y, f⟩
+             set rₗ : StructuredArrow one forgetReflexion :=
+                StructuredArrow.mk (Y := WalkingParallelPair.one) left
+             set rᵣ : StructuredArrow one forgetReflexion :=
+                StructuredArrow.mk (Y := WalkingParallelPair.one) right
+             have hrₗ : rₗ ∈ p
+             · suffices f : e₀ ⟶  rₗ by exact (h₂ f).mp h₁
+               exact StructuredArrow.homMk WalkingParallelPairHom.left
+             have hrᵣ : rᵣ ∈ p
+             · suffices f : e₀ ⟶  rᵣ by exact (h₂ f).mp h₁
+               exact StructuredArrow.homMk WalkingParallelPairHom.right
+             cases y <;> cases f
+             rotate_right 3
+             · exact h₁
+             · exact hrₗ
+             · exact hrᵣ
+             · set v : StructuredArrow one forgetReflexion :=
+                StructuredArrow.mk (Y := WalkingParallelPair.zero) left_reflexion
+               suffices f : v ⟶  rₗ by exact (h₂ f).mpr hrₗ
+               exact StructuredArrow.homMk (WalkingParallelPairHom.left)
+             · set v : StructuredArrow one forgetReflexion :=
+                StructuredArrow.mk (Y := WalkingParallelPair.zero) right_reflexion
+               suffices f : v ⟶  rᵣ by exact (h₂ f).mpr hrᵣ
+               exact StructuredArrow.homMk (WalkingParallelPairHom.right)
+
+variable {A B : C}
+
+/-- Bundle the data of a parallel pair along with a common section as a functor out of the walking
+reflexive pair -/
+def reflexivePair (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B) (sr : s ≫ g = 𝟙 B) :
+    (WalkingReflexivePair ⥤ C) where
+  obj x :=
+    match x with
+    | zero => B
+    | one => A
+  map h :=
+    match h with
+    | WalkingReflexivePairHom.id _ => 𝟙 _
+    | left => f
+    | right => g
+    | reflexion => s
+    | right_reflexion => g ≫  s
+    | left_reflexion => f ≫  s
+  map_comp := by
+    rintro _ _ _ ⟨⟩ g <;> cases g <;> try {rfl} <;>
+    try {simp [← Category.assoc, sl, sr]; rfl} <;> try {simp}
+
+/-- (Noncomputably) bundle the data of a reflexive pair as a functor out of the walking reflexive
+pair -/
+noncomputable def ofIsReflexivePair (f g : A ⟶ B) [IsReflexivePair f g] :
```

Comment:
It seems this is a particular case of `reflexivePair`, so that it is not necessary to reproduce the construction. You may apply `reflexivePair` to `f`, `g` and `commonSection f g`.